app/src/main/kotlin/com/wire/android/util/CurrentScreenManager.kt

```diff
@@ -123,17 +123,34 @@ class CurrentScreenManager @Inject constructor(
     }
 
     override fun onDestinationChanged(controller: NavController, destination: NavDestination, arguments: Bundle?) {
-        val currentView = currentScreenState.value.toString()
-        AnonymousAnalyticsManagerImpl.stopView(currentView)
+        val currentScreenName = currentScreenName()
+        currentScreenName?.let {
+            AnonymousAnalyticsManagerImpl.stopView(it)
+        }
+
         val currentItem = destination.toDestination()
         currentScreenState.value = CurrentScreen.fromDestination(
             currentItem,
             arguments,
             isApplicationVisibleFlow.value
         )
 
-        val newView = currentScreenState.value.toString()
-        AnonymousAnalyticsManagerImpl.recordView(newView)
+        val newScreenName = currentScreenName()
+        newScreenName?.let {
+            AnonymousAnalyticsManagerImpl.recordView(it)
+        }
+    }
+
+    private fun currentScreenName() = currentScreenState.value.let { currentScreen ->
+        when (currentScreen) {
+            is CurrentScreen.Home,
+            is CurrentScreen.Conversation,
+            is CurrentScreen.OtherUserProfile,
+            is CurrentScreen.ImportMedia,
+            is CurrentScreen.DeviceManager -> return@let currentScreen.toScreenName()
+
+            else -> return@let (currentScreen as? CurrentScreen.SomeOther)?.route
```

Comment:
lets never use route since it may contain sensitive informations (e.g. ids) 
we need to find different solutions line having this defined in the Destination like using the base route?